sources/packages/backend/apps/api/src/services/education-program/education-program.service.ts

```diff
@@ -119,13 +122,27 @@ export class EducationProgramService extends RecordDataModelService<EducationPro
         );
     }
 
+    if (educationProgram.sabcCode) {
+      const isSABCCodeDuplicate = await this.hasExistingProgramWithSameSABCCode(
+        institutionId,
+        educationProgram.sabcCode,
+        programId,
+      );
+      if (isSABCCodeDuplicate) {
+        throw new CustomNamedError("Duplicate SABC code.", DUPLICATE_SABC_CODE);
+      }
+    }
+
     // Assign attributes for update from payload only if existing program has no offering(s).
     if (!hasExistingOffering) {
       program.fieldOfStudyCode = educationProgram.fieldOfStudyCode;
       program.credentialType = educationProgram.credentialType;
       program.cipCode = educationProgram.cipCode;
       program.nocCode = educationProgram.nocCode;
-      program.sabcCode = educationProgram.sabcCode;
+      // Save SABC code as null in case of not answered in the form.
+      // This way it can be saved when multiple programs does not have a SABC code.
+      program.sabcCode =
```

Comment:
I missed it before but I believe it can be simplified as below:
```ts
program.sabcCode = program.sabcCode?.trim() || null;
```

Reply:
Ok! Done. Thanks!